src/test/java/net/openhft/chronicle/wire/JSONWireTest.java

```diff
@@ -519,4 +520,21 @@ private static void readList(SimpleTwoLists record, List<String> data, ValueIn r
             }
         }
     }
+
+    @Test
+    public void classReference() {
+        DtoWithClassReference dtoWithClassReference = new DtoWithClassReference();
+        dtoWithClassReference.implClass = this.getClass();
+        String json = JSON_ONLY.asString(dtoWithClassReference);
+        assertEquals("{\"@net.openhft.chronicle.wire.JSONWireTest$DtoWithClassReference\":{" +
+                        "\"implClass\":{\"@net.openhft.chronicle.wire.JSONWireTest\":{}}," +
```

Comment:
I don't think this is correct:
Class representation is now undistinguishable from empty object.
I suggest to change it to 
```json
"implClass":{"@type":"net.openhft.chronicle.wire.JSONWireTest"},
```